pandas_schema/validation.py

```diff
@@ -84,7 +84,10 @@ def get_errors(self, series: pd.Series, column: 'column.Column'):
         simple_validation = ~self.validate(series)
         if column.allow_empty:
             # Failing results are those that are not empty, and fail the validation
-            validated = (series.str.len() > 0) & simple_validation
+            if series.dtype == float:
```

Comment:
I suspect this should be `numpy.issubdtype(series.dtype, numpy.floating)`, but I could be wrong